python/lsst/daf/butler/script/queryCollections.py

```diff
@@ -47,7 +51,11 @@ def queryCollections(repo, collection_type, flatten_chains, include_chains):
         collection names.
     """
     butler = Butler(repo)
+    kwargs = {}
+    if glob:
+        kwargs['expression'] = globToRegex(glob)
```

Comment:
From your earlier version handling of `...` for this did make me wonder whether there is an efficiency gain for globToRegex to spot `*` as a specil case -- it probably should warn if `*` is passed in with other globs and if they just pass in `*` it should probably return an empty list rather than a regex if we assume that empty list means no constraints (that would save a lot of unnecessary regex processing when we know there is no filtering).